<!--

@license Apache-2.0

Copyright (c) 2024 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# eol-open-bracket-spacing

> [ESLint rule][eslint-rules] to enforce that no spaces are present between an opening parenthesis or bracket and a nested object or array expression at the end of a line.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var rule = require( '@stdlib/_tools/eslint/rules/eol-open-bracket-spacing' );
```

#### rule

[ESLint rule][eslint-rules] to enforce that no spaces are present between an opening parenthesis or bracket and a nested object or array expression at the end of a line.

**Bad**:

<!-- eslint-disable stdlib/eol-open-bracket-spacing -->

```javascript
var log = require( '@stdlib/console/log' );

log( {
    'foo': true
});

log( [
    1,
    2,
    3
]);

log( [ {
    'bar': true
}]);
```

**Good**:

```javascript
var log = require( '@stdlib/console/log' );

log({
    'foo': true
});

log([
    1,
    2,
    3
]);

log([{
    'bar': true
}]);
```

</section>

<!-- /.usage -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var Linter = require( 'eslint' ).Linter;
var rule = require( '@stdlib/_tools/eslint/rules/eol-open-bracket-spacing' );

var linter = new Linter();

var code = [
    'function test() {',
    '  var log = require( \'@stdlib/console/log\' );',
    '  log( {',
    '    "key": "value"',
    '  });',
    '  var arr = [ {',
    '    "nested": true',
    '  }];',
    '  log( arr );',
    '}'
].join( '\n' );

linter.defineRule( 'eol-open-bracket-spacing', rule );

var result = linter.verify( code, {
    'rules': {
        'eol-open-bracket-spacing': 'error'
    }
});
/* returns
    [
        {
            'ruleId': 'eol-open-bracket-spacing',
            'severity': 2,
            'message': 'No spaces allowed between an opening parenthesis or bracket and a nested object or array expression at the end of a line',
            'line': 3,
            'column': 3,
            'nodeType': 'CallExpression'
        },
        {
            'ruleId': 'eol-open-bracket-spacing',
            'severity': 2,
            'message': 'No spaces allowed between an opening parenthesis or bracket and a nested object or array expression at the end of a line',
            'line': 6,
            'column': 13,
            'nodeType': 'ArrayExpression'
        }
    ]
*/
```

</section>

<!-- /.examples -->

<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->

<section class="related">

</section>

<!-- /.related -->

<!-- Section for all links. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="links">

[eslint-rules]: https://eslint.org/docs/developer-guide/working-with-rules

</section>

<!-- /.links -->
